### PR TITLE
berkeley-abc: Replace with yosys-abc.

### PR DESCRIPTION
**Pull request by Aloz1**
yosys-abc is a fork of berkeley-abc by the YosysHQ team. The YosysHQ team advise that yosys should be used with yosys-abc instead of berkeley-abc. The two are compatible, with yosys-abc being better maintained.

Within Void berkeley-abc's only rdep is yosys. The berkeley-abc package has not been updated in some time. Rather than keeping both packages, this commit replaces berkeley-abc with yosys-abc.

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-glibc

### CODE REVIEW

**Comment by Aloz1** [comment]
Associated with change #52979

**Comment by sgn** [comment]
That's fine.
I would remove the `berkeley-abc` package, the `reverts` and replace `berkeley-abc` with `yosys-abc` in the `yosys` package.

File: srcpkgs/yosys-abc/template [hunk 1]
```diff
@@ -1,23 +1,22 @@
-# Template file for 'berkeley-abc'
-pkgname=berkeley-abc
-# This is the version shipped by Fedora 34
-_gitrev=d4fb192575cd319de68764eac5b7e049f575f43e
-version=1.01+20201124
+# Template file for 'yosys-abc'
+pkgname=yosys-abc
+reverts="1.01+20201124_1"
```
Comment (sgn) [inline srcpkgs/yosys-abc/template:3]: Just drop this.

File: srcpkgs/yosys-abc/template [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
 short_desc="System for Sequential Synthesis and Verification"
-maintainer="Đoàn Trần Công Danh <congdanhqx@gmail.com>"
+maintainer="Aloz1 <kno0001@gmail.com>"
 license="custom:University-of-California, BSD-4-Clause, MIT, BSD-2-Clause"
-homepage="http://www.eecs.berkeley.edu/~alanmi/abc/"
-distfiles="https://github.com/berkeley-abc/abc/archive/$_gitrev.tar.gz"
-checksum=52f1e9bf62c1e0d3b3cbe145f61b409a725e73fe606ba010efcbb2701a14171b
+homepage="http://github.com/YosysHQ/abc"
+distfiles="https://github.com/YosysHQ/yosys/releases/download/${version}/abc.tar.gz"
+checksum=e0a4eb6e4ec9209f7b81f7ad28ea0be1361a9cf4f4f639356feefc1f4450db7e
```
Comment (sgn) [inline srcpkgs/yosys-abc/template:15]: Adding:
```
replaces="berkeley-abc>=0"
provides="berkeley-abc-1.01+20210000_1"
```

File: srcpkgs/yosys-abc/template [hunk 3]
```diff
@@ -35,3 +34,9 @@ do_install() {
 	sed -n '/Copyright/,/SUCH DAMAGE/p' src/sat/bsat2/pstdint.h >pstdint.license
 	vlicense pstdint.license
 }
+
+berkeley-abc_package() {
+	build_style=meta
+	depends="${sourcepkg}>=${version}_${revision}"
+	short_desc+=" - transitional dummy package"
+}
```
Comment (sgn) [inline srcpkgs/yosys-abc/template:42]: And drop this.

I think noone would install `berkeley-abc` anyway.

**Comment by Aloz1** [comment]
> That's fine. I would remove the `berkeley-abc` package, the `reverts` and replace `berkeley-abc` with `yosys-abc` in the `yosys` package.

Sure thing. I'll update accordingly.

With the suggested changes, will an `xbps-install -Su` correctly uninstall berkeley-abc on systems that may have had it installed as a dependency of yosys?

**Comment by Aloz1** [comment]
@sgn I think I've addressed all your comments. Is there anything further you'd like me to address?

**Pull request merged**